src/AlgoliaSearch/Index.php

```
@@ -409,10 +409,159 @@ public function deleteByQuery($query, $args = array(), $waitLastCall = true)
+                        $i--;
+                    }
+                }
+            }
+
+            // iterate on facet numericFilters and remove the filter of the current disjunctive facet
+            for ($i = 0; $i < count($numericFilters); $i++) {
+                $needle = $facetName;
+
+                if (is_array($numericFilters[$i])) {
+                    foreach ($numericFilters[$i] as $filter) {
+                        if (substr($filter, 0, strlen($needle)) === $needle) {
+                            unset($numericFilters[$i]);
+                            $numericFilters = array_values($numericFilters);
+                            $i--;
+                            break;
+                        }
+                    }
+                } else {
+                    if (substr($numericFilters[$i], 0, strlen($needle)) === $needle) {
+                        unset($numericFilters[$i]);
+                        $numericFilters = array_values($numericFilters);
+                        $i--;
+                    }
+                }
+            }
+
+            $additionalParams = array(
+                'hitsPerPage' => 1,
+                'page' => 0,
+                'attributesToRetrieve' => array(),
+                'attributesToHighlight' => array(),
+                'attributesToSnippet' => array()
+            );
+
+
+            $additionalParams['facetFilters'] = $facetFilters;
+            $additionalParams['numericFilters'] = $numericFilters;
+
+            $queriesParams[$facetName] = array_merge($params, $additionalParams);
+        }
+
+        return $queriesParams;
+    }
+
+    /**
+     * @param $query
+     * @param $args
+     * @return mixed
+     * @throws AlgoliaException
+     */
+    private function searchWithDisjunctiveFaceting($query, $args)
+    {
+        if (isset($args['filters'])) {
+            throw new AlgoliaException('You can not use disjunctive faceting and the filters parameter');
```

We should throw \InvalidArgument exceptions in those cases and keep AlgoliaExceptions in the future for API related Exceptions.
Otherwise it makes it hard to know if we are dealing with runtime exceptions or logic exceptions.